### PR TITLE
User/colinli02/11 publish ros package actions final

### LINKED ISSUE #11: Publish ROS package actions
### Purpose

Each action that is used across our ROS packages should be published so that updates can be pushed to all ROS package repositories automatically.

### Changes

- Create repository for lint action
- Create repository for test action
- Use published actions in this repository

### Final To do
- [ ] Merge sailbot_workspace PR without switching branches
- [ ] Manually delete branch and re-check "Automatically delete head branches"
- [ ] Point network_systems PR to sailbot_workspace main branch, updated required jobs, then merge
- [ ] Create sailbot_workspace PR to point to main branch

### Resources

- https://docs.github.com/en/actions/creating-actions

### PR DESCRIPTION
**Pull request by colinli02**
### Description

- Resolves #11 
**⚠️ Two relevant branches, since modifications to 2 repos: ⚠️**
   - (This branch linked in PR from network_systems) + [sailbot_workspace branch](https://github.com/UBCSailbot/sailbot_workspace/tree/user/colinli02/11-Publish-ROS-package-actions_FINAL)

### Verification

- When running its CI, you can notice within "Actions" on both workspaces that it works, example:
[//this one calls itself](https://github.com/UBCSailbot/sailbot_workspace/actions/runs/4271545841 )
[//this one calls sailbot_workspace](https://github.com/UBCSailbot/network_systems/actions/runs/4271539418/jobs/7436072982 )

### Implementation
When this is merged into main, make sure within these files on line 22:
image

[Network_systems relevant yml file](https://github.com/UBCSailbot/network_systems/blob/user/colinli02/11-Publish-ROS-package-actions_FINAL/.github/workflows/call_sailbot_workspace_CI.yml)
[Sailbot_workspace relevant yml file](https://github.com/UBCSailbot/sailbot_workspace/blob/user/colinli02/11-Publish-ROS-package-actions_FINAL/.github/workflows/call_sailbot_workspace_CI.yml)
to switch to @main instead. I currently have it set to call my branches for verification, if you want to check.

### CODE REVIEW

**Comment by patrick-5546** [comment]
> ### Implementation
> When this is merged into main, make sure within these files on line 22: image
> 
> [Network_systems relevant yml file](https://github.com/UBCSailbot/network_systems/blob/user/colinli02/11-Publish-ROS-package-actions_FINAL/.github/workflows/call_sailbot_workspace_CI.yml) [Sailbot_workspace relevant yml file](https://github.com/UBCSailbot/sailbot_workspace/blob/user/colinli02/11-Publish-ROS-package-actions_FINAL/.github/workflows/call_sailbot_workspace_CI.yml) to switch to @main instead.

I think this would be the best way to merge everything in:

1. Merge in sailbot_workspace PR
    1. I temporarily disabled "automatically delete head branches" so that even the main branch CI will pass
2. Make and merge in a PR in sailbot_workspace to use main branch
3. Update this PR to use main branch

File: .github/workflows/call_sailbot_workspace_CI.yml
Comment (patrick-5546) [inline .github/workflows/call_sailbot_workspace_CI.yml:24]: Generalize so that this can be used in any subteam repo: I think `${{ github.event.repository.name }}` should work?

**Review by patrick-5546 — CHANGES_REQUESTED** [review]
Great work on this PR

General notes:
- Once the CI names have been finalized, you can update the required CI in https://github.com/UBCSailbot/network_systems/settings/branch_protection_rules/30406190

File: .github/workflows/call_sailbot_workspace_CI.yml
Comment (patrick-5546) [reply .github/workflows/call_sailbot_workspace_CI.yml:24]: Or `${{ github.repository }}`, not sure which one is better

Comment (patrick-5546) [reply .github/workflows/call_sailbot_workspace_CI.yml:24]: Actually it should match https://github.com/UBCSailbot/sailbot_workspace/blob/d9429e7bbdde97c618e32c391a51f4ddf7d58ef4/.github/workflows/tests.yml#L27, which currently uses `${{ github.repository }}`

Comment (patrick-5546) [reply .github/workflows/call_sailbot_workspace_CI.yml:24]: Actually actually see https://github.com/UBCSailbot/sailbot_workspace/pull/100#discussion_r1118139809

File: .github/workflows/call_sailbot_workspace_CI.yml
Comment (colinli02) [reply .github/workflows/call_sailbot_workspace_CI.yml:1]: Re-named to tests.yml

File: .github/workflows/call_sailbot_workspace_CI.yml
Comment (colinli02) [reply .github/workflows/call_sailbot_workspace_CI.yml:11]: Resolved. Run conditions is now:
```
# Run condition
on:
  push:
    branches:
      - main
  pull_request:
  workflow_dispatch:

  
 ```

File: .github/workflows/call_sailbot_workspace_CI.yml
Comment (colinli02) [reply .github/workflows/call_sailbot_workspace_CI.yml:7]: Replaced line 7 with
```
Runs the Tests workflow in sailbot_workspace
```

File: .github/workflows/call_sailbot_workspace_CI.yml
Comment (colinli02) [reply .github/workflows/call_sailbot_workspace_CI.yml:16]: Resolved, now:
```
# Run condition
on:
  push:
    branches:
      - main
  pull_request:
  workflow_dispatch:
  ```

File: .github/workflows/call_sailbot_workspace_CI.yml
Comment (colinli02) [reply .github/workflows/call_sailbot_workspace_CI.yml:25]: Added empty line to end of yml file.

File: .github/workflows/call_sailbot_workspace_CI.yml
Comment (colinli02) [reply .github/workflows/call_sailbot_workspace_CI.yml:20]: Replaced with
```
https://github.com/UBCSailbot/sailbot_workspace
```

File: .github/workflows/call_sailbot_workspace_CI.yml
Comment (colinli02) [reply .github/workflows/call_sailbot_workspace_CI.yml]: Done, used 
```
${{ github.event.repository.name }}
```

I made two separate commits, one with this and one without, can do a git revert in case ${{ github.event.repository.name }} doesn't work.

**Comment by colinli02** [comment]
This seems to appear, so checkout looks fine. Is this on my end?
image

**Review by patrick-5546 — APPROVED** [review]

**Comment by patrick-5546** [comment]
All that's left is to
- Run CI after the sailbot_workspace PR is merged in
- Get approval from @hhenry01

**Review by hhenry01 — APPROVED** [review]

**Pull request merged**
